### ISSUE TITLE
Repetitive execution of N4BiasFieldCorrection leads to unexpected behavior

### ISSUE THREAD

**Issue by romangrothausmann** [InsightSoftwareConsortium/ITK#1850]
### Description

It seems that repetitive execution of N4BiasFieldCorrection, e.g. if connected to SliceBySliceImageFilter, leads to unexpected behavior, see: 
https://discourse.itk.org/t/n4biasfieldcorrection-issues-when-re-run-applied-through-slicebyslice-filter/2897/8

### Steps to Reproduce

E.g. with this [n4_SBS](https://github.com/romangrothausmann/ITK-CLIs/blob/ee2fb980983a94eda5/n4_SBS.cxx)

### Expected behavior

That N4 yields the same output for the same input, independent of how many times it ran before on other images, i.e. no ramp effect as here:
![c706b5ff89c757af0af6cea18a76bc4af0c7fac4_2_493x499](https://user-images.githubusercontent.com/10707398/83972643-a48c7d80-a8e1-11ea-9523-b57fd02f6365.png)

### Actual behavior

Former runs of N4 seem to affect the result of later runs.

### Reproducibility

100%

### Versions

ITK @ 8e047cc46b0a58b32c420344ad98032cdb3d0fd0
https://github.com/romangrothausmann/ITK-CLIs/blob/ee2fb980983a94eda5/Dockerfile#L23

### Environment

As defined by 
https://github.com/romangrothausmann/ITK-CLIs/blob/ee2fb980983a94eda5/Dockerfile

### Additional Information

Apart from debugging this issue, it might be good to implement a new test for N4 that runs at least twice as suggested (https://discourse.itk.org/t/n4biasfieldcorrection-issues-when-re-run-applied-through-slicebyslice-filter/2897/10).
I also have the impression that the current test for N4 only tests writing the B-Spline bias field
https://github.com/InsightSoftwareConsortium/ITK/blob/d6d17b00986043efdd4b450a41a447e11c68b101/Modules/Filtering/BiasCorrection/test/itkN4BiasFieldCorrectionImageFilterTest.cxx#L312-L319
but not its application to the original image like
https://github.com/InsightSoftwareConsortium/ITKMontage/blob/2e7000149dd615dfb65d0664ec41a4026b95df11/examples/CompleteMontage.cxx#L284-L296
where as that example does not use `ReconstructBiasField` available since 
https://github.com/InsightSoftwareConsortium/ITK/commit/d6d17b00986043efdd4b450a41a447e11c68b101.

**Comment by romangrothausmann** [comment]
Possibly related:
The test for my [n4](https://github.com/romangrothausmann/ITK-CLIs/blob/d8b3b452c69bfda88bdea1d1615efdb255cdf246/n4.cxx)
https://github.com/romangrothausmann/ITK-CLIs/blob/d8b3b452c69bfda88bdea1d1615efdb255cdf246/.gitlab-ci.yml#L332-L337
is the only case where the output seems to depend on the architecture used for running the test, i.e. the md5sums of the output differ for our own runner
https://gitlab.com/romangrothausmann/ITK-CLIs/-/jobs/584777736
in contrast to when run on a shared GL-CI runner
https://gitlab.com/romangrothausmann/ITK-CLIs/-/jobs/584777243
@ntustison did you observe such dependence before?

**Comment by ntustison** [comment]
@romangrothausmann  No, I haven't noticed such dependence but my usage is pretty much limited to conventional MRI scans.  It's been other users such as yourself, and @blowekamp with his histology images, that have exercised usage which I have never tested.

**Comment by romangrothausmann** [comment]
@dzenanz @thewtex do you have an idea what could cause the hardware architecture dependence here?

**Comment by thewtex** [comment]
@romangrothausmann not sure -- floating point precision or multi-threading is often a source of variations, though...

**Comment by romangrothausmann** [comment]
Yes, that is what I would expect too, but oddly the n4-test is the only one that gives a different result (fails) if run on GL-CI runners, even though there are quite a few tests which use multi-threading and streaming and I would also expect that some depend on floating point precision.
